services/security/keyvault/key/wssd.go

```diff
@@ -86,6 +89,185 @@ func (c *client) CreateOrUpdate(ctx context.Context, group, vaultName, name stri
+	if err != nil {
+		return
+	}
+
+	// Private key wrapping info mandatory for both Import and Export
+	if parsedImportExportParams.PrivateKeyWrappingInfo == nil {
+		err = errors.Wrapf(errors.InvalidInput, "Private key wrapping info - Missing")
+		return
+	}
+
+	// Parse the key wrapping algorithm
+	keyWrappingAlgo, err = GetMOCKeyWrappingAlgorithm(*parsedImportExportParams.PrivateKeyWrappingInfo.KeyWrappingAlgorithm)
+
+	return
+}
+
+// Validate Export params
+func ParseAndValidateExportParams(keyValue *string, exportKey *wssdcloudsecurity.Key) (err error) {
+	parsedExportParams, keyWrappingAlgo, err := ParseAndValidateImportExportParams(keyValue)
+	if err != nil {
+		return err
+	}
+
+	if parsedExportParams.PrivateKeyWrappingInfo.PublicKey == nil {
+		// Wrapping public key mandatory for Export
+		return errors.Wrapf(errors.InvalidInput, "Wrapping public key - Missing")
+	}
+	wrappingKeyPublic, err := base64.URLEncoding.DecodeString(*parsedExportParams.PrivateKeyWrappingInfo.PublicKey)
+	if err != nil {
+		return err
+	}
+	wrappingKeyName := ""
+	if parsedExportParams.PrivateKeyWrappingInfo.KeyName != nil {
+		wrappingKeyName = *parsedExportParams.PrivateKeyWrappingInfo.KeyName
+	}
+
+	exportKey.PrivateKeyWrappingInfo = &wssdcloudsecurity.PrivateKeyWrappingInfo{
+		WrappingKeyName:   wrappingKeyName,
+		WrappingKeyPublic: wrappingKeyPublic,
+		WrappingAlgorithm: keyWrappingAlgo}
+	return
+}
+
+// Validate Import params
+func ParseAndValidateImportParams(keyValue *string, importKey *wssdcloudsecurity.Key) (err error) {
+	parsedImportParams, keyWrappingAlgo, err := ParseAndValidateImportExportParams(keyValue)
+	if err != nil {
+		return err
+	}
+
+	if parsedImportParams.PublicKey == nil {
+		return errors.Wrapf(errors.InvalidInput, "Public key - Missing")
+	}
+	importKey.PublicKey, err = base64.URLEncoding.DecodeString(*parsedImportParams.PublicKey)
+	if err != nil {
+		return err
+	}
+
+	if parsedImportParams.PrivateKey == nil {
+		return errors.Wrapf(errors.InvalidInput, "Private key - Missing")
+	}
+	importKey.PrivateKey, err = base64.URLEncoding.DecodeString(*parsedImportParams.PrivateKey)
+	if err != nil {
+		return err
+	}
+
+	if parsedImportParams.PrivateKeyWrappingInfo.KeyName == nil {
+		return errors.Wrapf(errors.InvalidInput, "Wrapping key name - Missing")
+	}
+	wrappingKeyName := *parsedImportParams.PrivateKeyWrappingInfo.KeyName
+
+	importKey.PrivateKeyWrappingInfo = &wssdcloudsecurity.PrivateKeyWrappingInfo{
+		WrappingKeyName:   wrappingKeyName,
+		WrappingAlgorithm: keyWrappingAlgo}
+	return
+}
+
+// Import
+func (c *client) ImportKey(ctx context.Context, group, vaultName, name string, param *keyvault.Key) (*keyvault.Key, error) {
+	err := c.validate(ctx, group, vaultName, name, param)
+	if err != nil {
+		return nil, err
+	}
+	if param.KeySize == nil {
+		return nil, errors.Wrapf(errors.InvalidInput, "Invalid KeySize - Missing")
+	}
+	request, err := getKeyRequest(wssdcloudcommon.Operation_IMPORT, group, vaultName, name, param)
```

Comment:
This function is okay, but instead of using Operation_IMPORT, can you reuse Operation_POST